R/generate_dust.R

```diff
@@ -806,32 +824,9 @@ generate_dust_assignment <- function(eq, name_state, dat, options = list()) {
     res <- sprintf("%s = %s;", lhs, rhs)
     is_array <- !is.null(eq$lhs$array)
     if (is_array) {
-      for (idx in rev(eq$lhs$array)) {
```

Comment:
This is moved into its own helper so that we can reuse it generating the expressions for the comparison